sw/device/lib/crypto/aes_gcm/aes_gcm.h

```diff
@@ -35,6 +35,39 @@ aes_error_t aes_gcm_ghash(const aes_block_t *hash_subkey,
                           const size_t input_len, const uint8_t *input,
                           aes_block_t *output);
 
+/**
+ * AES-GCM authenticated encryption as defined in NIST SP800-38D, algorithm 4.
+ *
+ * The key is represented as two shares which are XORed together to get the
+ * real key value. The IV must be either 96 or 128 bits.
+ *
+ * The byte-lengths of the plaintext and AAD must each be < 2^32. This is a
+ * tighter constraint than the length limits in section 5.2.1.1.
+ *
+ * Note that the above are bit lengths; the actual lengths passed to this
```

Comment:
I think "the above are bit lengths" might be out of date, since line 44 says "The byte-lengths of the plaintext...".